File: admin_panel/lib/features/users/interest_labels.dart
// ─────────────────────────────────────────────────────────────────────
// InterestLabels — admin paneldagi bola qiziqishlari uchun ID → label/icon
// ─────────────────────────────────────────────────────────────────────
//
// Bola child appda onboarding'da tanlagan ID'lar backend'da `interests
// String[]` sifatida saqlanadi. Admin paneldagi profilda chip ko'rinishida
// ko'rsatish uchun shu fayl mapping beradi.
//
// Manba (truth source) — farzandim_child/lib/features/onboarding/data/
// interest_options.dart. Ikkala ro'yxat qo'lda sinxron tutiladi.

import 'package:flutter/material.dart';

class InterestMeta {
  const InterestMeta({required this.label, required this.icon});
  final String label;
  final IconData icon;
}

const Map<String, InterestMeta> kAdminInterestLabels = {
  'book': InterestMeta(label: 'Kitoblar', icon: Icons.menu_book_rounded),
  'cartoon':
      InterestMeta(label: 'Multfilmlar', icon: Icons.movie_filter_rounded),
  'sport': InterestMeta(label: 'Sport', icon: Icons.sports_soccer_rounded),
  'music': InterestMeta(label: 'Musiqa', icon: Icons.music_note_rounded),
  'animals': InterestMeta(label: 'Hayvonlar', icon: Icons.pets_rounded),
  'space': InterestMeta(label: 'Kosmos', icon: Icons.rocket_launch_rounded),
  'tech': InterestMeta(label: 'Texnologiya', icon: Icons.memory_rounded),
  'art': InterestMeta(label: "San'at", icon: Icons.palette_rounded),
  'science': InterestMeta(label: 'Fan', icon: Icons.science_rounded),
  'games': InterestMeta(label: "O'yinlar", icon: Icons.sports_esports_rounded),
  'travel': InterestMeta(label: 'Sayohat', icon: Icons.map_rounded),
  'food': InterestMeta(label: 'Ovqat', icon: Icons.restaurant_rounded),
};

/// Noma'lum id uchun fallback (backend yangi tag qo'shgan, admin panel
/// hali yangilanmagan holatlarda).
InterestMeta labelFor(String id) {
  return kAdminInterestLabels[id] ??
      InterestMeta(label: id, icon: Icons.label_outline_rounded);
}

File: admin-web/src/lib/interest-labels.ts
// ─────────────────────────────────────────────────────────────────────
// Interest labels — bola onboarding'da tanlagan ID → label/icon
// ─────────────────────────────────────────────────────────────────────
//
// Backend `Child.interests` ID'larini admin paneldagi chip ko'rinishida
// foydalanuvchiga tanish so'zlar (Kitoblar, Multfilmlar...) bilan
// ko'rsatish uchun.
//
// Truth source — farzandim_child onboarding interest_options.dart. Ikkala
// ro'yxat qo'lda sinxron tutiladi (12 ta variant).

import type { LucideIcon } from 'lucide-react';
import {
  BookOpen,
  Film,
  Trophy,
  Music,
  Cat,
  Rocket,
  Cpu,
  Palette,
  FlaskConical,
  Gamepad2,
  Map as MapIcon,
  UtensilsCrossed,
  Tag,
} from 'lucide-react';

export interface InterestMeta {
  label: string;
  Icon: LucideIcon;
}

const INTEREST_LABELS: Record<string, InterestMeta> = {
  book: { label: 'Kitoblar', Icon: BookOpen },
  cartoon: { label: 'Multfilmlar', Icon: Film },
  sport: { label: 'Sport', Icon: Trophy },
  music: { label: 'Musiqa', Icon: Music },
  animals: { label: 'Hayvonlar', Icon: Cat },
  space: { label: 'Kosmos', Icon: Rocket },
  tech: { label: 'Texnologiya', Icon: Cpu },
  art: { label: "San'at", Icon: Palette },
  science: { label: 'Fan', Icon: FlaskConical },
  games: { label: "O'yinlar", Icon: Gamepad2 },
  travel: { label: 'Sayohat', Icon: MapIcon },
  food: { label: 'Ovqat', Icon: UtensilsCrossed },
};

/// Noma'lum id uchun fallback (backend yangi tag qo'shgan, admin panel
/// hali yangilanmagan holatlarda).
export function interestMetaFor(id: string): InterestMeta {
  return INTEREST_LABELS[id] ?? { label: id, Icon: Tag };
}
